### PR TITLE
fix(cli): validate observation percentage inputs

### LINKED ISSUE #4: observation add/update silently clamp percentage fields instead of validating input
## Problem

`observation add` and `observation update` document percentage inputs as 0-100 (`confidence`, `evidence`, `novelty`, `uncertainty`), but the CLI only checks integer format. Out-of-range values are silently clamped later in store logic.

So a command like:

```bash
obsxa observation add --project p1 --title t --source s --confidence 999
```

succeeds and stores `100` instead of failing fast.

## Why this matters

Silent coercion hides user mistakes. In scripts, this can produce misleading data quality because invalid inputs look successful.

## Affected paths

- `src/commands/observation.ts` (`parseOptionalInt`, add/update handlers)
- Store behavior in `src/core/observation.ts` (`clampPercent` usage)

## Suggested fix

- Enforce 0-100 in CLI argument parsing for these fields
- Keep clamping in core as defensive safety, but reject invalid CLI input with a clear error
- Add CLI tests for out-of-range values

### PR DESCRIPTION
**Pull request by oritwoen**
Closes #4

I hit this while checking machine-output workflows: passing values like "--confidence 999" looked successful but silently stored clamped values. This patch makes the CLI reject invalid percentage inputs up front so scripts fail loudly instead of writing misleading data.

What changed:
- enforce 0-100 bounds for observation add and update percentage fields (confidence, evidence, novelty, uncertainty)
- reject explicit empty values and unsafe integers in the shared observation integer parser
- add focused tests for missing, valid, out-of-range, empty, and unsafe integer inputs

Validation run locally:
- pnpm test:run
- pnpm lint
- pnpm typecheck
- pnpm build
- cubic review (no issues)
- coderabbit review --plain --type uncommitted (no findings)

### CODE REVIEW

File: src/commands/observation.ts [hunk 1]
```diff
@@ -116,17 +139,17 @@ export default defineCommand({
...
                 capturedAt: args["captured-at"],
                 sourceRef: args["source-ref"],
                 collector: args.collector,
                 inputHash: args["input-hash"],
-                evidenceStrength: parseOptionalInt(args.evidence, "evidence"),
-                novelty: parseOptionalInt(args.novelty, "novelty"),
-                uncertainty: parseOptionalInt(args.uncertainty, "uncertainty"),
+                evidenceStrength: parseOptionalInt(args.evidence, "evidence", percentRange),
+                novelty: parseOptionalInt(args.novelty, "novelty", percentRange),
+                uncertainty: parseOptionalInt(args.uncertainty, "uncertainty", percentRange),
```
[the remark this answers is not in the document]
Comment (oritwoen) [reply src/commands/observation.ts]: a3f0e64
